src/ci/github-actions/ci.py

```diff
@@ -181,30 +183,76 @@ def format_run_type(run_type: WorkflowRunType) -> str:
         raise AssertionError()
 
 
+def run_workflow_locally(job_data: Dict[str, Any], job_name: str):
+    DOCKER_DIR = Path(__file__).absolute().parent.parent / "docker"
+
+    jobs = list(job_data["auto"])
+    jobs.extend(job_data["pr"])
+
+    jobs = [job for job in jobs if job.get("image") == job_name]
+    if len(jobs) == 0:
+        raise Exception(f"Job `{job_name}` not found")
+    job = jobs[0]
```

Comment:
It can pick a job with the same name if a PR and auto job have the same name, which can happen. I ordered the jobs so that auto jobs have priority; I didn't want to complicate the CLI by allowing users to select between PR and auto jobs. But it's true that they might not actually have the same config even if they have the same name, so maybe we should let users choose.